mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java

```diff
@@ -0,0 +1,170 @@
+
+    @Inject
+    public TikaTextExtractor(TikaHttpClient tikaHttpClient) {
+        this.tikaHttpClient = tikaHttpClient;
+        this.objectMapper = initializeObjectMapper();
+    }
+
+    private ObjectMapper initializeObjectMapper() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        SimpleModule mapModule = new SimpleModule();
+        mapModule.addDeserializer(ContentAndMetadata.class, new ContentAndMetadataDeserializer());
+        objectMapper.registerModule(mapModule);
+        return objectMapper;
+    }
+
+    @Override
+    public ParsedContent extractContent(InputStream inputStream, String contentType, String fileName) throws Exception {
+        ContentAndMetadata contentAndMetadata = convert(tikaHttpClient.metadataAndContent(inputStream, contentType));
+        return new ParsedContent(contentAndMetadata.getContent(), contentAndMetadata.getMetadata());
+    }
+
+    private ContentAndMetadata convert(String tika) throws IOException, JsonParseException, JsonMappingException {
+        return objectMapper.readValue(tika, ContentAndMetadata.class);
+    }
+
+    private static class ContentAndMetadataDeserializer extends JsonDeserializer<ContentAndMetadata> {
+
+        @Override
+        public ContentAndMetadata deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
+            TreeNode treeNode = jsonParser.getCodec().readTree(jsonParser);
+            Preconditions.checkState(treeNode.isArray() && treeNode.size() == 1, "The response should have only one element");
+            Preconditions.checkState(treeNode.get(0).isObject(), "The element should be a Json object");
+            ObjectNode node = (ObjectNode) treeNode.get(0);
+            return ContentAndMetadata.from(ImmutableList.copyOf(node.fields()).stream()
+                .map(entry -> Pair.of(entry.getKey(), asListOfString(entry.getValue())))
+                .collect(Collectors.toMap(Pair::getLeft, Pair::getRight)));
+        }
+
+        private List<String> asListOfString(JsonNode jsonNode) {
+            if (jsonNode.isArray()) {
+                return ImmutableList.copyOf(jsonNode.elements()).stream()
+                    .map(node -> node.asText())
+                    .collect(Collectors.toList());
+            }
+            return ImmutableList.of(jsonNode.asText());
+        }
+        
+    }
+    
+    private static class ContentAndMetadata {
+
+        private static final String TIKA_HEADER = "X-TIKA";
+        private static final String CONTENT_METADATA_HEADER_NAME = TIKA_HEADER + ":content";
+
+        public static ContentAndMetadata from(Map<String, List<String>> contentAndMetadataMap) {
+            String onlySpaces = null;
+            return new ContentAndMetadata(content(contentAndMetadataMap, onlySpaces),
+                    Maps.filterEntries(contentAndMetadataMap, new Predicate<Map.Entry<String, List<String>>>() {
```

Comment:
It should be easier for reading if extract Predicate to constant or function I think